src/hdmf_zarr/backend.py

```diff
@@ -531,11 +547,18 @@ def __resolve_ref(self, zarr_ref):
             try:
                 target_zarr_obj = target_zarr_obj[object_path]
             except Exception:
-                raise ValueError("Found bad link to object %s in file %s" % (object_path, source_file))
+                # TODO: Find why the reference includes the root object.
+                # This is a fix to allow the target object to be found
+                try:
+                    object_path = pathlib.Path(object_path)
+                    rel_obj_path = object_path.relative_to(*object_path.parts[:2])
+                    target_zarr_obj = target_zarr_obj[rel_obj_path]
+                except Exception:
+                    raise ValueError("Found bad link to object %s in file %s" % (object_path, source_file))
```

Comment:
With the fix in `__get_ref` using `export_source`, this workaround on read should no longer be necessary

Reply:
@oruebel we do need this still. The source is correct, but there actual path includes the source. This is a fix to remove the source. This would be a bandage until the next update on the backend to flush out export, but also the paths for references.

Reply:
That seems like an issue that should be addressed in `__get_ref`. We are correcting the `source` in `__get_ref`. We'll probably need to do something similar for `path`. We shouldn't create files with bad paths.